tools/doc/html.js

```diff
@@ -382,13 +382,19 @@ function buildToc({ filename, apilinks }) {
       node.children.push({ type: 'html', value: anchor });
     });
 
-    file.toc = unified()
-      .use(markdown)
-      .use(gfm)
-      .use(remark2rehype, { allowDangerousHtml: true })
-      .use(raw)
-      .use(htmlStringify)
-      .processSync(toc).toString();
+    if (toc !== '') {
+      file.toc = '<details id="toc" open><summary>Table of contents</summary>' +
+        unified()
+          .use(markdown)
+          .use(gfm)
+          .use(remark2rehype, { allowDangerousHtml: true })
+          .use(raw)
+          .use(htmlStringify)
+          .processSync(toc).toString() +
+        '</details>';
+    } else {
+      file.toc = '<!-- TOC -->';
```

Comment:
Alternatively, we could rollback to `<h2>` for the Index:

```suggestion
      file.toc = '<div id="toc"><h2>Table of contents</h2></div>';
```

It would let us revert the changes in `tools/doc/allhtml.js`, but also means the `all.html` page would not have a retractable TOC anymore.

Reply:
`all.js` is the most useful place for this feature so I'm inclined to do the extra work to keep the `<details>` tag there.